components/channel_view/channel_view.jsx

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import $ from 'jquery';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import * as UserAgent from 'utils/user_agent.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+import deferComponentRender from 'components/deferComponentRender';
+
+import ChannelHeader from 'components/channel_header';
+import CreatePost from 'components/create_post.jsx';
+import FileUploadOverlay from 'components/file_upload_overlay.jsx';
+import PostView from 'components/post_view';
+import TutorialView from 'components/tutorial/tutorial_view.jsx';
+
+export default class ChannelView extends React.Component {
+    static propTypes = {
+
+        /**
+         * ID of the channel to display
+         */
+        channelId: PropTypes.string.isRequired,
+
+        /**
+         * Set if this channel is deactivated, primarily used for DMs with inactive users
+         */
+        deactivatedChannel: PropTypes.bool.isRequired,
+
+        /**
+         * Set to show the tutorial
+         */
+        showTutorial: PropTypes.bool.isRequired,
+
+        /**
+         * React router params
+         */
+        params: PropTypes.object
```

Comment:
What do we need from the params here? We're already passing the channelId and whether or not it's the tutorial as other props. If we get rid of this, we can switch it to a PureComponent

Reply:
Hmm, good question. I assumed it was being used for something in component since it was previously defined but I don't see it being used at all. I'll remove it